controller/linodemachine_controller.go

```diff
@@ -369,41 +367,55 @@ func (r *LinodeMachineReconciler) reconcileCreate(
 	return ctrl.Result{}, nil
 }
 
-func (r *LinodeMachineReconciler) configureDisksControlPlane(
+func (r *LinodeMachineReconciler) configureDisks(
 	ctx context.Context,
 	logger logr.Logger,
 	machineScope *scope.MachineScope,
 	linodeInstanceID int,
 ) error {
-	if !kutil.IsControlPlaneMachine(machineScope.Machine) {
+	if machineScope.LinodeMachine.Spec.DataDisks == nil && machineScope.LinodeMachine.Spec.OSDisk != nil {
 		return nil
 	}
 
-	if err := r.resizeRootDisk(ctx, logger, machineScope, linodeInstanceID); err != nil {
+	err := r.resizeRootDisk(ctx, logger, machineScope, linodeInstanceID)
+	if err != nil {
 		return err
 	}
-
-	if !reconciler.ConditionTrue(machineScope.LinodeMachine, ConditionPreflightEtcdDiskCreated) {
-		// create the etcd disk
-		if _, err := machineScope.LinodeClient.CreateInstanceDisk(
-			ctx,
-			linodeInstanceID,
-			linodego.InstanceDiskCreateOptions{
-				Label:      "etcd-data",
-				Size:       defaultEtcdDiskSize,
-				Filesystem: string(linodego.FilesystemExt4),
-			},
-		); err != nil {
-			logger.Error(err, "Failed to create etcd disk")
-
-			conditions.MarkFalse(machineScope.LinodeMachine, ConditionPreflightEtcdDiskCreated, string(cerrs.CreateMachineError), clusterv1.ConditionSeverityWarning, err.Error())
-
+	if !reconciler.ConditionTrue(machineScope.LinodeMachine, ConditionPreflightAdditionalDisksCreated) {
+		for deviceName, disk := range machineScope.LinodeMachine.Spec.DataDisks {
+			if disk.DiskID != 0 {
+				continue
+			}
+			label := disk.Label
+			if label == "" {
+				label = deviceName
+			}
+			// create the disk
+			linodeDisk, err := machineScope.LinodeClient.CreateInstanceDisk(
+				ctx,
+				linodeInstanceID,
+				linodego.InstanceDiskCreateOptions{
+					Label:      label,
+					Size:       disk.SizeMB,
+					Filesystem: string(linodego.FilesystemExt4),
+				},
+			)
+			if err != nil {
+				logger.Error(err, "Failed to create disk", "DiskLabel", label)
+
+				conditions.MarkFalse(machineScope.LinodeMachine, ConditionPreflightAdditionalDisksCreated, string(cerrs.CreateMachineError), clusterv1.ConditionSeverityWarning, err.Error())
```

Comment:
nit: line length consistency

```suggestion
				conditions.MarkFalse(
					machineScope.LinodeMachine, 
					ConditionPreflightAdditionalDisksCreated, 
					string(cerrs.CreateMachineError), 
					clusterv1.ConditionSeverityWarning,
					err.Error(),
				)
```